presto-main/src/test/java/com/facebook/presto/sql/planner/TestPredicatePushdown.java

```diff
@@ -60,4 +61,30 @@ public void testPushDownToLhsOfSemiJoin()
                                                         "LINE_QUANTITY", "quantity")))),
                                 anyTree(tableScan("orders", ImmutableMap.of("ORDERS_ORDER_KEY", "orderkey"))))));
     }
+
+    @Test(enabled = false) // unignore after fixing #10085
+    public void testNonDeterministicPredicatePropagatesOnlyToSourceSideOfSemiJoin()
+    {
+        assertPlan("SELECT * FROM lineitem WHERE orderkey IN (SELECT orderkey FROM orders) AND orderkey = random(5)",
+                anyTree(
+                        semiJoin("LINE_ORDER_KEY", "ORDERS_ORDER_KEY", "SEMI_JOIN_RESULT",
+                                anyTree(
+                                        filter("LINE_ORDER_KEY = CAST(random(5) AS bigint)",
+                                                        tableScan("lineitem", ImmutableMap.of(
+                                                                "LINE_ORDER_KEY", "orderkey")))),
+                                anyTree(
+                                        project(// NO filter here
```

Comment:
how do you know? Filter could be above the project.

See `com.facebook.presto.sql.planner.assertions.PlanMatchPattern#anyNot`, maybe we need something like `anyTreeNot`

Reply:
I change the `anyTree` to explicit `node(ExchangeNode.class`